packages/build/tests/plugins/tests.js

```diff
@@ -807,6 +809,81 @@ test('Pinning plugin versions takes into account the compatibility field', async
   })
 })
 
+const runWithPluginRunsMock = async function (
+  t,
+  fixture,
+  { flags, status, sendStatus = true, testPlugin, pluginRuns = DEFAULT_TEST_PLUGIN_RUNS } = {},
+) {
+  const { scheme, host, requests, stopServer } = await startServer([
+    { path: PLUGIN_RUNS_PATH, response: pluginRuns, status },
+    { path: PLUGINS_LIST_URL, response: getPluginsList(testPlugin), status: 200 },
+  ])
+  try {
+    await runFixture(t, fixture, {
+      flags: {
+        siteId: 'test',
+        token: 'test',
+        sendStatus,
+        testOpts: { scheme, host, pluginsListUrl: `${scheme}://${host}` },
+        ...flags,
+      },
+    })
+  } finally {
+    await stopServer()
+  }
+  t.snapshot(requests)
+}
+
+const PLUGIN_RUNS_PATH = `/api/v1/sites/test/plugin_runs/latest`
+
+test('Pin netlify.toml-only plugin versions', async (t) => {
+  await runWithPluginRunsMock(t, 'pin_config_success')
+})
+
+test('Does not pin netlify.toml-only plugin versions if there are no matching plugin runs', async (t) => {
+  await runWithPluginRunsMock(t, 'pin_config_success', { pluginRuns: [{ package: `${TEST_PLUGIN_NAME}-test` }] })
+})
+
+test('Does not pin netlify.toml-only plugin versions if there are no plugin runs', async (t) => {
+  await runWithPluginRunsMock(t, 'pin_config_success', { pluginRuns: [] })
+})
+
+test('Does not pin netlify.toml-only plugin versions if there are no matching plugin runs version', async (t) => {
+  await runWithPluginRunsMock(t, 'pin_config_success', { pluginRuns: [{ package: TEST_PLUGIN_NAME }] })
+})
+
+// Node 8 `util.inspect()` prints the API error differently
+// @todo remove after removing support for Node 8
+if (!version.startsWith('v8.')) {
+  test('Fails the build when pinning netlify.toml-only plugin versions and the API request fails', async (t) => {
```

Comment:
Should we `not pin` instead of failing due to an API request failure? We can try and pin it next build? 
Unless we expect it always to fail

Reply:
From discussing with @vbrown608, if the API call to fetch the latest plugin runs fail, we decided that it'd be better to fail the build rather than run the build with a potentially wrong version of the plugin. 

Considering API calls would most likely fail when there is some API outage, and builds cannot work anyway during those outage, this seems like a sensible approach to me, but I'm curious to hear what you think?

Reply:
Good point that makes sense. thanks for clarifying.